types/api/v1beta1/event.pb.go

```diff
@@ -0,0 +1,1680 @@
+}
+
+func (x *Parent) Reset() {
+	*x = Parent{}
+	if protoimpl.UnsafeEnabled {
+		mi := &file_types_api_v1beta1_event_proto_msgTypes[4]
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		ms.StoreMessageInfo(mi)
+	}
+}
+
+func (x *Parent) String() string {
+	return protoimpl.X.MessageStringOf(x)
+}
+
+func (*Parent) ProtoMessage() {}
+
+func (x *Parent) ProtoReflect() protoreflect.Message {
+	mi := &file_types_api_v1beta1_event_proto_msgTypes[4]
+	if protoimpl.UnsafeEnabled && x != nil {
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		if ms.LoadMessageInfo() == nil {
+			ms.StoreMessageInfo(mi)
+		}
+		return ms
+	}
+	return mi.MessageOf(x)
+}
+
+// Deprecated: Use Parent.ProtoReflect.Descriptor instead.
+func (*Parent) Descriptor() ([]byte, []int) {
+	return file_types_api_v1beta1_event_proto_rawDescGZIP(), []int{4}
+}
+
+func (x *Parent) GetExecutable() *Executable {
+	if x != nil {
+		return x.Executable
+	}
+	return nil
+}
+
+func (x *Parent) GetPid() int32 {
+	if x != nil {
+		return x.Pid
+	}
+	return 0
+}
+
+func (x *Parent) GetEntityId() uint32 {
+	if x != nil {
+		return x.EntityId
+	}
+	return 0
+}
+
+func (x *Parent) GetNamespacedPid() int32 {
+	if x != nil && x.NamespacedPid != nil {
+		return *x.NamespacedPid
+	}
+	return 0
+}
+
+func (x *Parent) GetRealUser() *User {
+	if x != nil {
+		return x.RealUser
+	}
+	return nil
+}
+
+func (x *Parent) GetThread() *Thread {
+	if x != nil {
+		return x.Thread
+	}
+	return nil
+}
+
+func (x *Parent) GetParent() *Process {
+	if x != nil {
+		return x.Parent
+	}
+	return nil
+}
+
+type Executable struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	Name string `protobuf:"bytes,1,opt,name=name,proto3" json:"name,omitempty"`
+}
+
+func (x *Executable) Reset() {
+	*x = Executable{}
+	if protoimpl.UnsafeEnabled {
+		mi := &file_types_api_v1beta1_event_proto_msgTypes[5]
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		ms.StoreMessageInfo(mi)
+	}
+}
+
+func (x *Executable) String() string {
+	return protoimpl.X.MessageStringOf(x)
+}
+
+func (*Executable) ProtoMessage() {}
+
+func (x *Executable) ProtoReflect() protoreflect.Message {
+	mi := &file_types_api_v1beta1_event_proto_msgTypes[5]
+	if protoimpl.UnsafeEnabled && x != nil {
+		ms := protoimpl.X.MessageStateOf(protoimpl.Pointer(x))
+		if ms.LoadMessageInfo() == nil {
+			ms.StoreMessageInfo(mi)
+		}
+		return ms
+	}
+	return mi.MessageOf(x)
+}
+
+// Deprecated: Use Executable.ProtoReflect.Descriptor instead.
+func (*Executable) Descriptor() ([]byte, []int) {
+	return file_types_api_v1beta1_event_proto_rawDescGZIP(), []int{5}
+}
+
+func (x *Executable) GetName() string {
+	if x != nil {
+		return x.Name
+	}
+	return ""
+}
+
+type User struct {
+	state         protoimpl.MessageState
+	sizeCache     protoimpl.SizeCache
+	unknownFields protoimpl.UnknownFields
+
+	Id uint32 `protobuf:"varint,1,opt,name=id,proto3" json:"id,omitempty"`
```

Comment:
I see that all the fields are generated with json omitempty.
But for some fields we don't want to omit if empty.
For example here - uid 0 is root, and we definitely don't want to drop it.

Reply:
Oh, true.. We can't have primitives for this semantic than, because 0 will be dropped, best to use a wrapper https://protobuf.dev/reference/java/api-docs/com/google/protobuf/UInt32Value, than if not set it is actually nil, instead of 0. WDTY?